gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ObservationCommentMapper.java

```diff
@@ -0,0 +1,148 @@
+        II narrativeStatementId, List<Encounter> encounters) {
+        var composition =
+            EhrResourceExtractorUtil.extractEhrCompositionForNarrativeStatement(ehrExtract, narrativeStatementId);
+
+        encounters
+            .stream()
+            .filter(encounter -> encounter.getId().equals(composition.getId().getRoot()))
+            .findFirst()
+            .ifPresent(encounter -> observation.setContext(new Reference(encounter)));
+    }
+
+    private void setObservationComment(Observation observation, String text) {
+        if (!text.isBlank()) {
+            observation.setComment(text.trim());
+        }
+    }
+
+    private Meta createMeta() {
+        var meta = new Meta();
+        meta.setProfile(Collections.singletonList(new UriType(META_URL)));
+
+        return meta;
+    }
+
+    private Identifier createIdentifier(String narrativeStatementId) {
+        var identifier = new Identifier();
+        identifier.setSystem(IDENTIFIER_SYSTEM);
+        identifier.setValue(narrativeStatementId);
+
+        return identifier;
+    }
+
+    private InstantType createIssued(RCMRMT030101UK04EhrExtract ehrExtract, II narrativeStatementId) {
+        RCMRMT030101UK04EhrComposition composition =
+            EhrResourceExtractorUtil.extractEhrCompositionForNarrativeStatement(ehrExtract, narrativeStatementId);
+
+        if (composition.getAuthor().getTime().getNullFlavor() == null) {
+            return DateFormatUtil.parseToInstantType(composition.getAuthor().getTime().getValue());
+        }
+
+        return DateFormatUtil.parseToInstantType(ehrExtract.getAvailabilityTime().getValue());
+    }
+
+    private Reference createPerformer(RCMRMT030101UK04EhrExtract ehrExtract, RCMRMT030101UK04NarrativeStatement narrativeStatement) {
+        RCMRMT030101UK04EhrComposition composition =
+            EhrResourceExtractorUtil.extractEhrCompositionForNarrativeStatement(ehrExtract, narrativeStatement.getId());
+
+        return ParticipantReferenceUtil.getParticipantReference(narrativeStatement.getParticipant(), composition);
+    }
+
+    private CodeableConcept createCodeableConcept() {
+        var codeableConcept = new CodeableConcept();
+        codeableConcept.setCoding(
+            Collections.singletonList(new Coding(CODING_SYSTEM, CODING_CODE, CODING_DISPLAY)));
+
+        return codeableConcept;
+    }
+
+    private List<RCMRMT030101UK04EhrComposition> getCompositionsWithNarrativeStatement(RCMRMT030101UK04EhrExtract ehrExtract) {
+        return ehrExtract.getComponent()
+            .stream()
+            .flatMap(component -> component.getEhrFolder().getComponent().stream())
+            .map(RCMRMT030101UK04Component3::getEhrComposition)
+            .filter(ehrComposition -> ehrComposition.getComponent()
+                .stream()
+                .map(RCMRMT030101UK04Component4::getNarrativeStatement)
+                .anyMatch(Objects::nonNull))
```

Comment:
Remember you need to filter out any NarrativeStatements that contain an element with a referredToExternalDocument element eg.

`<reference typeCode="REFR">
<referredToExternalDocument classCode="DOC" moodCode="EVN">
                <id root="394559384658936"/>
                <code code="37251000000104" codeSystem="2.16.840.1.113883.2.1.3.2.4.15" displayName="Other digital signal"/>
                <text mediaType="text/plain">
                    <reference value="file://localhost/AbsentAttachment394559384658936.txt"/>
                </text>
            </referredToExternalDocument>
        </reference>`

Reply:
Ah yes, thank you for spotting this.